crates/re_build_search_index/src/ingest/docs.rs

```diff
@@ -0,0 +1,69 @@
+use std::path::Path;
+
+use super::{Context, DocumentData, DocumentKind};
+
+pub fn ingest(ctx: &Context) -> anyhow::Result<()> {
+    let progress = ctx.progress_bar("docs");
+
+    let dir = ctx.workspace_root().join("docs").join("content");
+    for entry in glob::glob(&format!("{dir}/**/*.md"))? {
+        let entry = entry?;
+        let path = entry
+            .strip_prefix(&dir)?
+            .with_extension("")
+            .display()
+            .to_string();
+        progress.set_message(path.clone());
+        let url = format!("https://rerun.io/docs/{path}");
+        let (frontmatter, body) = parse_docs_frontmatter(&entry)?;
+
+        ctx.push(DocumentData {
+            kind: DocumentKind::Docs,
+            title: frontmatter.title,
+            hidden_tags: vec![],
+            tags: vec![],
+            content: body,
+            url,
+        });
+    }
+
+    ctx.finish_progress_bar(progress);
+
+    Ok(())
+}
+
+#[derive(serde::Deserialize)]
+struct DocsFrontmatter {
+    title: String,
+}
+
+fn parse_docs_frontmatter<P: AsRef<Path>>(path: P) -> anyhow::Result<(DocsFrontmatter, String)> {
+    const START: &str = "---";
+    const END: &str = "---";
+
+    let path = path.as_ref();
+    let content = std::fs::read_to_string(path)?;
+
+    let Some(start) = content.find(START) else {
+        anyhow::bail!("{:?} is missing frontmatter", path.display())
+    };
+    let start = start + START.len();
+
+    let Some(end) = content[start..].find(END) else {
+        anyhow::bail!(
+            "{:?} has invalid frontmatter: missing {END:?} terminator",
+            path.display()
+        );
+    };
+    let end = start + end;
+
+    let frontmatter: DocsFrontmatter =
+        serde_yaml::from_str(content[start..end].trim()).map_err(|err| {
```

Comment:
Are these the frontmatter of e.g. `docs/content/reference/types.md`? What yaml features are we using, and could we be using toml instead?

Reply:
Yes, we're using the same features as in examples, and we could also use TOML here

Reply:
https://github.com/rerun-io/rerun/issues/5203